### ISSUE TITLE
Saving array of ObjectIds in a model association does not respect array order

### ISSUE THREAD

**Issue by dinana** [balderdashy/sails-mongo#151]
If 2 models have a many-to-many association and I save an array of model 1 ObjectIds to model2 the array order is not respected.

Update: The issue might be in retrieval not insertion, in recent tests, the records are inserted in-order in the relationship collection, but upon retrieval order is not respected.

**Comment by rmckeel** [comment]
I have the same issue.  Boiled down, I can change the order, and it saves as expected to MongoDB as an ordered array.  I can then change the order again.  
```
b.as.splice(to, 0, b.as.splice(from, 1)[0]);
b.save(...)
```

When I retrieve results via the blueprint API myserver/b, it shows as (the plural of 'a') in the same order every time.

I could add some more logic to the app and model, but I'd really like to figure out how to get Sails to keep the order of the array that is in the database.

in a.js:
```
    b : {
        model: 'b'
    }
```
in b.js:
```
    as : {
      collection: 'a',
      via: 'b'
    }
```

**Comment by sailsbot** [comment]
Thanks for posting, @dinana.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
